From e5ac768c35b8f03fd89b107c78e9c35328ad27c1 Mon Sep 17 00:00:00 2001
From: Seth Forshee <seth.forshee@canonical.com>
Date: Mon, 27 Jul 2015 09:16:54 -0500
Subject: [PATCH 0087/1767] UBUNTU: SAUCE: overlayfs: Enable user namespace
 mounts for the "overlay" fstype

User namespace mounts were mistakenly only enabled for the
"overlayfs" fstype and not for the "overlay" fstype. This is a
regression in behavior towards userspace from previous Ubuntu
kernels and breaks some features of LXC.

BugLink: http://bugs.launchpad.net/bugs/1478578
Signed-off-by: Seth Forshee <seth.forshee@canonical.com>
Acked-by: Chris J Arges <chris.j.arges@canonical.com>
Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
---
 fs/overlayfs/super.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fs/overlayfs/super.c b/fs/overlayfs/super.c
index 8b2369c..d510975 100644
--- a/fs/overlayfs/super.c
+++ b/fs/overlayfs/super.c
@@ -1174,6 +1174,7 @@ static struct file_system_type ovl_fs_type = {
 	.name		= "overlay",
 	.mount		= ovl_mount,
 	.kill_sb	= kill_anon_super,
+	.fs_flags	= FS_USERNS_MOUNT,
 };
 MODULE_ALIAS_FS("overlay");
 
-- 
1.9.3

